File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingCalendarBreakColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingCalendarBreakRow } from "./TrainingCalendarBreakRow";

export interface TrainingCalendarBreakColumns {
    TrainingCalendarBreakId: Column<TrainingCalendarBreakRow>;
    Title: Column<TrainingCalendarBreakRow>;
    TrainingCalendarName: Column<TrainingCalendarBreakRow>;
    StartDate: Column<TrainingCalendarBreakRow>;
    EndDate: Column<TrainingCalendarBreakRow>;
    IsActive: Column<TrainingCalendarBreakRow>;
}

export class TrainingCalendarBreakColumns extends ColumnsBase<TrainingCalendarBreakRow> {
    static readonly columnsKey = 'Administration.TrainingCalendarBreak';
    static readonly Fields = fieldsProxy<TrainingCalendarBreakColumns>();
}

File: QadeerApp.Web/Modules/Administration/TrainingCalendarBreak/TrainingCalendarBreakColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingCalendarBreak")]
[BasedOnRow(typeof(TrainingCalendarBreakRow), CheckNames = true)]
public class TrainingCalendarBreakColumns
{
    [EditLink, AlignRight]
    public int TrainingCalendarBreakId { get; set; }

    [EditLink, Width(200)]
    public string Title { get; set; }

    [Width(200)]
    public string TrainingCalendarName { get; set; }

    [Width(120)]
    public DateTime StartDate { get; set; }

    [Width(120)]
    public DateTime EndDate { get; set; }

    [Width(90)]
    public bool IsActive { get; set; }
}
